package.json

```diff
@@ -62,17 +62,17 @@
   "devDependencies": {
     "@types/glob": "^7.1.3",
     "@types/mocha": "^8.2.2",
-    "@types/node": "^14.14.37",
+    "@types/node": "^18.0.0",
     "@types/vscode": "^1.55.0",
     "glob": "7.1.6",
-    "mocha": "8.3.2",
+    "mocha": "10.0.0",
     "prettier": "2.3.2",
     "ts-loader": "8.1.0",
     "typescript": "4.3.5",
-    "vsce": "1.87.1",
+    "vsce": "2.9.1",
     "vscode-debugadapter-testsupport": "1.46.0",
-    "webpack": "4.46.0",
-    "webpack-cli": "3.3.12"
+    "webpack": "5.40.0",
+    "webpack-cli": "4.10.0"
   },
```

Comment:
Did you confirm that these new dev dependencies, and their transitive dependencies, are still ALv2 compliant according to ASF?

Since they are dev dependencies I'd assume that means no part of them ends up in `extension.js` or the .vsix file (is this true?) and so we don't need to update the `bin.{LICNESE,NOTICE} files, but they do still need to be ALv2 compatible since they are required to build.

Just so it's clear to everyone:
* packages listed in `dependencies` and their transitive dependencies *must* be listed in `bin.LICENSE` or `bin.NOTICE`, with the assumption they end up in the vsix file
* packages listed in "devDependencies" and their transitive dependencies *should not* be listed anywhere, which the assumptino that they don't end up in the vsix file
* all packages need to be ALv2 compatible

I think we do this correctly, but I'm not positive. Is there a way to list the LICENSE/NOTICE information of all dependencies and their transitives that *aren't* devDependencies?

Reply:
I didn't mean to update the webpack items those were in part of testing the other two things that broke items I will revert those.

But yes your are correct the `devDependencies` do not end up in the VSIX but the items in `dependencies` do which is how we have everything setup. I also do know that all but one of the items under `devDependencies` are ALv2 compatible. Is the ISC license compatible with ALv2? `glob` (https://github.com/isaacs/node-glob) is the only package under `devDependencies` that uses something other MIT or AL

Reply:
Does updating webpack break other things? I was asking more generally about all the new dev deps. If that updates without any problem I would just keep it as is.

Here's the list of license approved by ASF: https://www.apache.org/legal/resolved.html

ISC is Category A which is allowed. Note that we also need to verify the transitive dependencies of the devDependencies. It's possible that one of our devDependencies depends on something that ASF doesn't allow.

Reply:
Nothing breaks with the current updates. All items are allowed under ALv2 but I am not sure about their attached deps so I will look into that

Reply:
@stevedlawrence Pretty handy tool
```
yarn global add license-checker
```
then you can run
```
license-checker --summary
```
Here's the result
```
├─ MIT: 253
├─ ISC: 38
├─ BSD-3-Clause: 20
├─ BSD-2-Clause: 16
├─ Apache-2.0: 11
├─ MIT*: 2
├─ Python-2.0: 1
├─ Custom: http://github.com/substack/node-bufferlist: 1 # <-- MIT
├─ CC-BY-4.0: 1
├─ (MIT OR WTFPL): 1
├─ (BSD-2-Clause OR MIT OR Apache-2.0): 1
└─ 0BSD: 1
```
These all are approved correct? Only one I wasn't sure about was `0BSD`

Reply:
The ASF licensing policy says BSD and variants are acceptable. Though, according to https://opensource.org/licenses/0BSD, 0BSD is basically ISC. Either way, 0BSD should be fine.

Sounds good then. No concerns with any of the licenses.